zdm/zdm.py

```diff
@@ -112,31 +113,74 @@ def vector_diff_power_law(Eth,*params):
     
     low=np.where(Eth < Emin)[0]
     if len(low) > 0:
-        result[low]=0.
+        result[low]=1.  # This was 0 and I think it was wrong -- JXP
     high=np.where(Eth > Emax)[0]
     if len(high) > 0:
         result[high]=0.
     
     return result
 
 
-############### unused - to delete ##########
-# power-laws here are differential
-#def power_law_norm(Emin,Emax,gamma):
-#	""" Calculates the normalisation factor for a power-law """
-#	return Emin**gamma-Emax**-gamma
+########### gamma functions #############
+
+def vector_cum_gamma(Eth,*params):
+    """ Calculates the fraction of bursts above a certain gamma function
+    for a given Eth.
+    """
+    params=np.array(params)
+    Emin=params[0]
+    Emax=params[1]
+    gamma=params[2]
+
+    # Calculate
+    norm = Emax*float(mpmath.gammainc(gamma, a=Emin/Emax))
+    Eth_Emax = Eth/Emax
+    # If this is too slow, we can adopt scipy + recurrance
```

Comment:
So scipy + recurrence means ? Would this mean first precalculating this integral on a grid of a, storing it, and using lookup tables? Or else one could treat the integral as two parts; one, for a <= a* << 1, where it is essentially a pure power law and hence trivially fast; and for a > a* where you care about it being a gamma function.

Reply:
I somewhat forget.  It refers to a comment on the intranets for speeding
up this type of integral.  Probably isn't worth it.